New icons

matthewthomsonnz opened the pull request:
This PR:
- Removes vanilla font awesome
- Adds custom (tree-shaked) font awesome + additional custom icons together via https://fontastic.me/
- https://fontastic.me/ login is in lastpass for future icon font changes
- Here are the 29 icons, pretty sure I haven't missed any:
![image](https://user-images.githubusercontent.com/32156362/162858214-017ad299-85fc-4699-83f4-d89d8622b900.png)


[AB#6845](https://dev.azure.com/S72/fefacba9-96b6-4af2-a53d-050ed453e13a/_workitems/edit/6845)

curtisadams28 commented:
@matthewthomsonnz I've replaced the pagination icon with a styled button. It's probably easiest if you remove all the carousel changes as they will just cause merge conflicts.

matthewthomsonnz commented:
> @matthewthomsonnz I've replaced the pagination icon with a styled button. It's probably easiest if you remove all the carousel changes as they will just cause merge conflicts.

Sounds good thanks. I've reverted the carousel changes and included fa-circle for the time being.

PeculiarGoat commented on site/templates/application/nav/user_logged_in.jet:
```diff
@@ -3,8 +3,8 @@
-          <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" class="user-icon" width="24" height="24" x="0px" y="0px" viewBox="-241 245 24 24" xml:space="preserve">
-            <path d="M-231,247c-5.5,0-10,4.5-10,10c0,5.5,4.5,10,10,10s10-4.5,10-10C-221,251.5-225.5,247-231,247 M-224.6,261.8c-1.4-1.7-4.9-2.3-6.4-2.3s-4.9,0.6-6.4,2.3c-1-1.3-1.6-3-1.6-4.8c0-4.4,3.6-8,8-8s8,3.6,8,8C-223,258.8-223.6,260.5-224.6,261.8M-231,251c-1.9,0-3.5,1.6-3.5,3.5s1.6,3.5,3.5,3.5s3.5-1.6,3.5-3.5S-229.1,251-231,251"/>
+          <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 5C13.66 5 15 6.34 15 8C15 9.66 13.66 11 12 11C10.34 11 9 9.66 9 8C9 6.34 10.34 5 12 5ZM12 19.2C9.5 19.2 7.29 17.92 6 15.98C6.03 13.99 10 12.9 12 12.9C13.99 12.9 17.97 13.99 18 15.98C16.71 17.92 14.5 19.2 12 19.2Z" fill="#F8F8F8"/>
```
Is that a hard-coded colour in there?

matthewthomsonnz replied:
- indeed it is - removed

PeculiarGoat replied:
I meant this one lol: `fill="#F8F8F8"`

PeculiarGoat replied:
Still got that hardcoded fill on the path element.

matthewthomsonnz replied:
- removed

PeculiarGoat commented on site/styles/_nav.scss:
```diff
@@ -397,7 +397,7 @@
     border-radius: 50%;
     color: rgba(var(--body-color-rgb), 0.7);
-    height: 32px;
+    font-size: 24px;
```
Looks like this gets overridden by the `.fa` class in font-awesome _core.scss.

matthewthomsonnz replied:
I'm not sure it does? When I remove this `font-size: 24px;` property the search close button gets smaller. Also I'm unsure where _core.scss is.

PeculiarGoat replied:
_core.scss was in font-awesome... not that you've removed it, this issue is resolved!

PeculiarGoat commented on site/styles/_share-modal.scss:
```diff
@@ -32,7 +36,14 @@
       left: -30px;
+
+      /* stylelint-disable selector-max-compound-selectors, max-nesting-depth */
+      svg {
```
This never gets applied to the Letterboxd icon because that svg has inline styles.

matthewthomsonnz replied:
- removed

PeculiarGoat commented on site/styles/_shift72.scss:
```diff
@@ -11,8 +11,12 @@
 .s72-btn-search {
   @extend .btn-primary-override;
+  font-size: 24px;
+  height: var(--input-height);
```
`height` doesn't seem to do anything here.

matthewthomsonnz replied:
- removed

PeculiarGoat commented on site/styles/_awards.scss:
```diff
@@ -27,7 +27,8 @@
 .meta-awards-icon-overlay {
   @extend .text-center;
-
+  @extend .d-flex;
```
Can you please replace these bootstrap classes with the actual CSS.

matthewthomsonnz replied:
- replaced these three

PeculiarGoat reviewed (changes requested):
Some things to look at.  Also, should the font-awesome dependency be removed if we are no longer using font-awesome fonts?

matthewthomsonnz commented:
> Some things to look at. Also, should the font-awesome dependency be removed if we are no longer using font-awesome fonts?

- Ah yes! Thanks. Removed font awesome from sass and package.json.

PeculiarGoat reviewed (changes requested):
Getting there.

A few points/questions to look at:

- Anywhere that has text directly next to an icon (like awards, availability label, social media buttons in share modal, etc), the text looks misaligned compared to the old icons.  Does this need to be addressed?  I notice that in the designs, some text has been removed entirely (see the wishlist button)... should that text be removed in this PR then?
- The trailer button doesn't have a new icon like in the designs... is there a reason for that?
- The custom icons (svg user account dropdown and the span hamburger :face_vomiting:) look like they are a different scale to the other icons... should they be replaced or modified as part of this task too?

matthewthomsonnz commented:
> * Anywhere that has text directly next to an icon (like awards, availability label, social media buttons in share modal, etc), the text looks misaligned compared to the old icons.  Does this need to be addressed?  I notice that in the designs, some text has been removed entirely (see the wishlist button)... should that text be removed in this PR then?

@PeculiarGoat It looks not too bad with my eye. Also take a look at my comparison testing below, it seems reasonable. Let me know specifically which ones look misaligned to you though and I will amend.
![Screenshot from 2022-04-26 11-11-08](https://user-images.githubusercontent.com/32156362/165189627-8860a81b-c29d-4a18-9f67-b984db958157.png)

I notice the wishlist button/icon is higher compared to before so will fix that now:
![Screenshot from 2022-04-26 11-10-03](https://user-images.githubusercontent.com/32156362/165189833-b721cc05-fff4-4c92-bcd8-f63897699013.png)

matthewthomsonnz commented:
> I notice that in the designs, some text has been removed entirely (see the wishlist button)... should that text be removed in this PR then?

You may recall I tried to do this and it got messy between the old core template versions and relish. A separate task has been specifically made to handle this separately. https://dev.azure.com/S72/SHIFT72/_workitems/edit/6478

> * The trailer button doesn't have a new icon like in the designs... is there a reason for that?

Is included with the above task as icon is svg hardcoded in relish

> * The custom icons (svg user account dropdown and the span hamburger :face_vomiting:) look like they are a different scale to the other icons... should they be replaced or modified as part of this task too?

- svg icon is done in this PR
- I know the spans are weird but it has a nice transition to a close button, keen to leave as is

PeculiarGoat commented:
Huh... maybe I'm going crazy?!  You are right they are mostly fine.

So yeah that wishlist one needs sorting out.

The only one I can think of that you haven't screenshot'd is the "Use a new card" icon in the purchase modal (visible when your account already has a saved card).  But the difference is really small.

Other than that... the icons in the share modal look a bit weird to me but apparently they did before and they do in the designs too... maybe that can be ignored?

matthewthomsonnz commented:
> Huh... maybe I'm going crazy?! You are right they are mostly fine.
> 
> So yeah that wishlist one needs sorting out.
> 
> The only one I can think of that you haven't screenshot'd is the "Use a new card" icon in the purchase modal (visible when your account already has a saved card). But the difference is really small.
> 
> Other than that... the icons in the share modal look a bit weird to me but apparently they did before and they do in the designs too... maybe that can be ignored?

Haha.Yeah that credit card icon was a touch higher, have adjusted (as well as the nav search close) slightly.

Add to wishlist adjusted.

I'm keen on ignored as they seem okay. I can always do a pixel perfect sweep PR later on if design complains.

PeculiarGoat reviewed (changes requested):
I just snuck in some css linting and a small change to the hamburger icon alignment.

Almost happy with this PR... just a couple more icons from Relish that need looking at:
- `fa-ellipsis-v` in the purchase modal needs alignment to match the [design](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/609b4324218908be34583c6e).  To find this icon you need to have a saved credit card on your account.  Then in the purchase modal click 'Use a new card' and the ellipsis will appear directly to the right of 'Save this card for future use' but it is currently styled to be the same colour as the background.  :(
- `fa-exclamation-circle` appears to be missing from the list on the ADO card.  It can be seen in the designs [here](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/609b4324218908be34583c6e) for the subscription change credit card modal on the subscriptions page.  It can probably just use the same icon as `fa-warning` right?
- `fa-file-text-o` is also missing from the list on the ADO card.  Can be seen in this [design](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/6205cb089be501bfd74ee4ff) for the subscriptions page title.
- `fa-exclamation-triangle` is used in the user subscription component in Relish, but I can't see it anywhere in the designs.

PeculiarGoat commented on site/styles/_forms.scss:
```diff
@@ -155,11 +155,16 @@ s72-activatedevice-form,
+
     .s72-btn {
-      color: $button-text-color;
+      color: var(--body-color);
```
You're going to have to add a specific case for the ellipsis because changing this colour and hover colour affects at least one other button:
![image](https://user-images.githubusercontent.com/10665859/165689889-7ac7236b-7397-4af7-a0ad-6d3a15ab0cd0.png)
Design team aren't going to like that lol

matthewthomsonnz replied:
- Fixed

PeculiarGoat reviewed (changes requested):
Okay last thing I promise!

curtisadams28 reviewed (approved):
LGTM. The account logo in the nav should probably be changed to use the icon font. That can be done in a seperate task though. I'll put it on my todo list.
